futures-core/src/future/mod.rs

```diff
@@ -8,6 +8,19 @@ pub use core::future::Future;
 mod future_obj;
 pub use self::future_obj::{FutureObj, LocalFutureObj, UnsafeFutureObj};
 
+/// A `Future` or `TryFuture` which tracks whether or not the underlying future
+/// should no longer be polled.
+///
+/// `is_terminated` will return `true` if a future should no longer be polled.
+/// Usually, this state occurs after `poll` (or `try_poll`) returned
+/// `Poll::Ready`. However, `is_terminated` may also return `false` if a future
```

Comment:
This `false` seems like it should be `true`?

Reply:
Yup (all the booleans got flipped around from an earlier version of this PR).

Reply:
Done.